language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/MakeConstructPublicCodeAction.java

```diff
@@ -0,0 +1,124 @@
+        if (!DIAGNOSTIC_CODE.contains(diagnostic.diagnosticInfo().code()) || context.currentSyntaxTree().isEmpty()
+                || context.currentSemanticModel().isEmpty()) {
+            return Collections.emptyList();
+        }
+        Range diagnosticRange = CommonUtil.toRange(diagnostic.location().lineRange());
+        NonTerminalNode nonTerminalNode = CommonUtil.findNode(diagnosticRange, context.currentSyntaxTree().get());
+        Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(nonTerminalNode);
+        if (symbol.isEmpty() || symbol.get().getModule().isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Optional<Project> project = context.workspace().project(context.filePath());
+        if (project.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Optional<Path> filePath = CommonUtil.getFilePathForSymbol(symbol.get(), project.get(), context);
+        if (filePath.isEmpty() || context.workspace().syntaxTree(filePath.get()).isEmpty()) {
+            return Collections.emptyList();
+        }
+        URI uri = filePath.get().toUri();
+
+        Optional<NonTerminalNode> node = CommonUtil.findNode(symbol.get(),
+                context.workspace().syntaxTree(filePath.get()).get());
+        if (node.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Optional<Position> position = getStartPosition(node.get());
+        if (position.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Range range = new Range(position.get(), position.get());
+        String editText = SyntaxKind.PUBLIC_KEYWORD.stringValue() + " ";
+        TextEdit textEdit = new TextEdit(range, editText);
+        List<TextEdit> editList = List.of(textEdit);
+        String commandTitle = String.format(CommandConstants.MAKE_CONSTRUCT_PUBLIC, symbol.get().getName().orElse(""));
+        return List.of(createCodeAction(commandTitle, editList, uri.toString(), CodeActionKind.QuickFix));
+    }
+
+    private static Optional<Position> getStartPosition(Node node) {
+        SyntaxKind typeKind = node.kind();
+
+        if (typeKind.equals(SyntaxKind.TYPE_DEFINITION)) {
+            TypeDefinitionNode typeDefinitionNode = (TypeDefinitionNode) node;
+            return Optional.of(CommonUtil.toPosition(typeDefinitionNode.typeKeyword().lineRange().startLine()));
+        }
+        if (typeKind.equals(SyntaxKind.CLASS_DEFINITION)) {
+            ClassDefinitionNode classDefinitionNode = (ClassDefinitionNode) node;
+            if (!classDefinitionNode.classTypeQualifiers().isEmpty()) {
+                return Optional.of(CommonUtil.toPosition(classDefinitionNode.classTypeQualifiers().get(0)
+                        .lineRange().startLine()));
+            } else {
+                return Optional.of(CommonUtil.toPosition(classDefinitionNode.classKeyword().lineRange().startLine()));
+            }
```

Comment:
If we consider the content within the if and else clauses, the differed logic is capturing the start line while the return statement can be taken out. So, let's get the start line to a variable and have a single return. WDYT?